### PR TITLE
Adding support for readonly model PVCs

### PR DESCRIPTION
**Pull request by visheshtanksale**

### CODE REVIEW

File: internal/controller/platform/standalone/nimservice_test.go [hunk 1]
```diff
@@ -114,7 +114,8 @@ var _ = Describe("NIMServiceReconciler for a standalone platform", func() {
 					},
 				},
 				NIMCache: appsv1alpha1.NIMCacheVolSpec{
-					Name: "test-nimcache",
+					Name:     "test-nimcache",
+					ReadOnly: true,
```
Comment (slu2011) [inline internal/controller/platform/standalone/nimservice_test.go:118]: How about keep the old testcase and assert readonly is false (default value) then add the new testcase such that readOnly is set to true?

**Review by slu2011 — DISMISSED** [review]

File: api/apps/v1alpha1/nimcache_types.go [hunk 2]
```diff
@@ -132,6 +132,7 @@ type PersistentVolumeClaim struct {
 	// VolumeAccessMode is the volume access mode of the PVC
 	VolumeAccessMode corev1.PersistentVolumeAccessMode `json:"volumeAccessMode,omitempty"`
 	SubPath          string                            `json:"subPath,omitempty"`
+	ReadOnly         bool                              `json:"readOnly,omitempty"`
```
Comment (shivamerla) [inline api/apps/v1alpha1/nimcache_types.go:135]: `ReadOnly` should not be part of NIMCache volume spec as we always need it to be rw for caching, instead we need to add it under `NIMService`.

Comment (visheshtanksale) [reply api/apps/v1alpha1/nimcache_types.go:135]: I have added `ReadOnly` to `NIMCacheVolSpec`. 
Its here on the PersistenceVolume to support use case in which user has created and populated the PVC without using NIMCache.

File: internal/controller/platform/standalone/nimservice_test.go (hunk 1 above)
Comment (visheshtanksale) [reply internal/controller/platform/standalone/nimservice_test.go]: Added more tests

File: api/apps/v1alpha1/nimcache_types.go (hunk 2 above)
Comment (visheshtanksale) [reply api/apps/v1alpha1/nimcache_types.go]: Fixed after update to the CRD

**Review by shivamerla — APPROVED** [review]
lgtm

**Pull request merged**